File: 数组array/双指针/easy-27-移除元素/removeElement.js
/**
* @param {number[]} nums
* @param {number} val
* @return {number}
*/
var removeElement = function(nums, val) {
    // 双指针	44 ms	41.2 MB	JavaScript
    const len = nums.length;
    let i = 0;
    for (let j = 0; j< len; j++) {
        if (nums[j] !== val) {
            nums[i] = nums[j];
            i++;
        }
    }
    return i;
};

File: 数组array/双指针/easy-27-移除元素/removeElement.go
func removeElement(nums []int, val int) int {
    // 双指针 0 ms	2 MB	Go
    i := 0
    for j:=0; j< len(nums); j++ {
        if nums[j] != val {
            nums[i] = nums[j]
            i++
        }
    }
    return i
}